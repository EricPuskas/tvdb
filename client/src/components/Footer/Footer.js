import React from "react";
import "./Footer.scss";

const Footer = () => {
  return (
    <div className="Footer" data-test="footer">
      <span> Copyright &copy; {new Date().getFullYear()} </span>
    </div>
  );
};

export default Footer;
