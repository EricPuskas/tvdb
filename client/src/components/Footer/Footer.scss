.Footer {
  color: #717171;
  width: 100%;
  text-align: center;
  font-size: 1rem;
  padding: 0.5rem;
  position: absolute;
  bottom: 0;
  left: 0;
}
