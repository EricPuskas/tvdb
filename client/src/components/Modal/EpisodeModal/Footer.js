import React from "react";

const Footer = props => {
  return (
    <div
      className="modal-footer text-center"
      style={{ paddingBottom: "1rem", borderTop: "1px solid #0000002e" }}
    >
      {props.children}
    </div>
  );
};

export default Footer;
