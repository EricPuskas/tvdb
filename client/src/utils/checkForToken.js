import jwt_decode from "jwt-decode";

// Redux Store
import store from "../store";

// Utilities
import setToken from "./setToken";

// Actions
import { getToken } from "../actions/token";

const checkForToken = () => {
  // Check for token
  if (!localStorage.token) {
    store.dispatch(getToken());
    setToken(localStorage.token);
  } else {
    setToken(localStorage.token);
    // Decode token to be able to check if it expired
    const decoded = jwt_decode(localStorage.token);
    // Check for expired token
    const currentTime = Date.now() / 1000;
    if (decoded.exp < currentTime) {
      store.dispatch(getToken());
    }
  }
};

export default checkForToken;
