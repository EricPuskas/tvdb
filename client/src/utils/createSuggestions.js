const filterInput = input => {
  return input.replace(/[^\w\s]/gi, "");
};

const createSuggestions = (word, data) => {
  let filter = filterInput(word);
  const re = new RegExp(`${filter.toLowerCase()}.*\\B`, "g");
  return data.filter(item => re.test(item.toLowerCase()));
};

export default createSuggestions;
