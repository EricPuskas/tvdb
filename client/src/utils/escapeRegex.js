const escapeRegex = text => {
  if (text !== "" || text !== null || text !== undefined) {
    // replace all white space with +, globally (replace all occurrences)
    return text.replace(/[\s]/g, "+");
  }
};

export default escapeRegex;
