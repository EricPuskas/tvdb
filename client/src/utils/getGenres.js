import React from "react";
import uuid from "uuid/v4";

const getGenres = genres => {
  if (!genres) return null;
  return genres.map(genre => (
    <li key={uuid()}>
      <span className="tag">{genre}</span>
    </li>
  ));
};

export default getGenres;
