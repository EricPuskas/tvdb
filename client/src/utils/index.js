import { applyMiddleware, createStore } from "redux";
import rootReducer from "../reducers";
import { middleware } from "../store";

export const findByAttribute = (component, attr) =>
  component.find(`[data-test="${attr}"]`);

export const testStore = initState => {
  const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
  return createStoreWithMiddleware(rootReducer, initState);
};
