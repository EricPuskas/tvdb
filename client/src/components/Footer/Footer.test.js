import React from "react";
import { shallow } from "enzyme";
import { findByAttribute } from "../../utils";
import Footer from "./Footer";

describe("Footer Component", () => {
  const component = shallow(<Footer />);
  const wrapper = findByAttribute(component, "footer");

  it("Should render without errors", () => {
    expect(wrapper.length).toBe(1);
  });

  it("Should pass snapshot test", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
