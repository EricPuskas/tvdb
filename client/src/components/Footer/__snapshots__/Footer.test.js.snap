// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component Should pass snapshot test 1`] = `"<div class=\\"Footer\\" data-test=\\"footer\\"><span> Copyright Â© 2019 </span></div>"`;
